File: app/src/main/java/org/androidforfun/droids/model/ShapeS.java
/*
 *  Copyright (C) 2016 Salvatore D'Angelo
 *  This file is part of Droids project.
 *
 *  Droids is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  Droids is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License.
 */
package org.androidforfun.droids.model;

/*
 * This is a S shape.
 */
public class ShapeS extends Shape {
    public ShapeS() {
        super(3, 2);
        rotation_block = blocks[1];
        rotation_cycle = 2;
    }

    // The x,y position of each block is adjusted according to
    // the shape position and size. The adjust is first applied
    // assuming the shape is its first position. Then it is rotated.
    // The color is green.
    public Block[] getBlocks() {
        blocks[0].setX(x + 2);
        blocks[0].setY(y);
        blocks[1].setX(x + 1);
        blocks[1].setY(y);
        blocks[2].setX(x + 1);
        blocks[2].setY(y + 1);
        blocks[3].setX(x);
        blocks[3].setY(y + 1);

        apply_rotation();

        for (Block block : blocks) {
            block.setColor(0xff00ff00);
        }
        return blocks;
    }
}

File: app/src/main/kotlin/org/code4projects/droids/model/ShapeS.kt
/*
 * Copyright (c) 2016-2026 Salvatore D'Angelo
 * Licensed under the MIT License. See the LICENSE file in the project root.
 */
package org.code4projects.droids.model

/*
 * This is a S shape.
 */
class ShapeS : Shape(3, 2) {
    init {
        rotationBlock = blockArray[1]
        rotationCycle = 2
    }

    // The x,y position of each block is adjusted according to
    // the shape position and size. The adjust is first applied
    // assuming the shape is its first position. Then it is rotated.
    // The color is green.
    override fun getBlocks(): Array<Block> {
        blockArray[0].x = x + 2
        blockArray[0].y = y
        blockArray[1].x = x + 1
        blockArray[1].y = y
        blockArray[2].x = x + 1
        blockArray[2].y = y + 1
        blockArray[3].x = x
        blockArray[3].y = y + 1

        applyRotation()

        for (block in blockArray) {
            block.color = 0xff00ff00L.toInt()
        }
        return blockArray
    }
}
